### PR TITLE
[13.0][MIG] partner_vat_unique: Migration to 13.0

### PR DESCRIPTION
**Pull request by manuelcalerosolis**
This is the migration to v 13.0 module apartner_vat_unique

@Tecnativa TT22085

### CODE REVIEW

File: partner_vat_unique/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from odoo.tools import config
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    vat = fields.Char(copy=False)
+
+    @api.constrains("vat")
+    def _check_vat_unique(self):
```
Comment (astirpe) [inline partner_vat_unique/models/res_partner.py:16]: In V13 there's this method [`def _compute_same_vat_partner_id(self)`](https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/res_partner.py#L261-L269) that looks quite similar.

File: partner_vat_unique/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# Copyright 2020 Manuel Calero - Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Partner VAT Unique",
+    "summary": "Module to make the VAT number unique for customers and suppliers.",
+    "version": "13.0.1.0.1",
```
Comment (ernestotejeda) [inline partner_vat_unique/__manifest__.py]:
```suggestion
    "version": "13.0.1.0.0",
```

File: partner_vat_unique/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#. module: partner_vat_unique
+#: model:ir.model,name:partner_vat_unique.model_res_partner
+msgid "Contact"
+msgstr "Contacto"
+
+#. module: partner_vat_unique
+#: model:ir.model.fields,field_description:partner_vat_unique.field_res_partner__vat
+#: model:ir.model.fields,field_description:partner_vat_unique.field_res_users__vat
+msgid "Tax ID"
+msgstr "Tax "
```
Comment (ernestotejeda) [inline partner_vat_unique/i18n/es.po]:
```suggestion
msgstr "Tax"
```

**Review by ernestotejeda — CHANGES_REQUESTED** [review]
@manuelcalerosolis You can generate the readme

File: partner_vat_unique/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# Copyright 2020 Manuel Calero - Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (NL66278) [inline partner_vat_unique/__manifest__.py]: Make it https:

File: partner_vat_unique/models/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (NL66278) [inline partner_vat_unique/models/res_partner.py]: https: everywhere....

File: partner_vat_unique/README.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    :alt: OCA/partner-contact
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_vat_unique
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/134/12.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
```
Comment (NL66278) [inline partner_vat_unique/README.rst:27]: Maybe explain somewhere that uniqueness is only enforced on the main, or commercial partner, of customers/suppliers/partners. As vat is a commercial field that is basically copied to the underlying partners.

**Review by NL66278 — CHANGES_REQUESTED** [review]
:+1: Looks good mostly but a few nags.

File: partner_vat_unique/i18n/es.po (hunk 3 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/i18n/es.po]: Changed

File: partner_vat_unique/models/res_partner.py (hunk 5 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/models/res_partner.py]: Changed

File: partner_vat_unique/__manifest__.py (hunk 4 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/__manifest__.py]: Done

File: partner_vat_unique/README.rst (hunk 6 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/README.rst:27]: Added this comments at INSTALL.rst

File: partner_vat_unique/models/res_partner.py (hunk 1 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/models/res_partner.py:16]: Now used the field "same_vat_partner_id", thanks.

File: partner_vat_unique/__manifest__.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_vat_unique/__manifest__.py]: This is still incorrect.

File: partner_vat_unique/README.rst [hunk 7]
```diff
@@ -39,13 +39,22 @@ Installation
 Will not check previous VAT duplicates, so it is recomended make sure there
 isn't any duplicated VAT before installation.
 
+Just verify the main contacts, as VAT is a commercial, field that is basically copied to the underlying contacts.
+
+Known issues / Roadmap
+======================
+
+* Do not allow the creation of duplicate VATs, you are currently notified but would let you create it.
```
Comment (pedrobaeza) [inline partner_vat_unique/README.rst]: This is not what we have talked. These are the problems with the standard one. This module prevents effectively this. The only know issues is when partners are shared between several companies.

File: partner_vat_unique/readme/INSTALL.rst [hunk 8]
```diff
@@ -1,2 +1,4 @@
 Will not check previous VAT duplicates, so it is recomended make sure there
 isn't any duplicated VAT before installation.
+
+Just verify the main contacts, as VAT is a commercial, field that is basically copied to the underlying contacts.
```
Comment (pedrobaeza) [inline partner_vat_unique/readme/INSTALL.rst]: Why are you putting this text in install section?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
See my comments

**Review by misern2 — CHANGES_REQUESTED** [review]
Functional Tested. Ok. Waiting to solve other request changes

**Comment by ernestotejeda** [comment]
ping @manuelcalerosolis

File: partner_vat_unique/__manifest__.py (hunk 2 above)
Comment (manuelcalerosolis) [reply partner_vat_unique/__manifest__.py]: Corrected!

**Review by ernestotejeda — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@NL66278 can you please update your review?

**Review by NL66278 — APPROVED** [review]
:+1: Code Review

**Pull request merged**